Extra translations + fixups

LiamBrenner opened the pull request:
Fix for #4.

LiamBrenner committed:
Marked some strings for translation

LiamBrenner committed:
Added django messages

timheap commented:
cc @frague59. Neither @SableWalnut nor I have done much with translations, so your input on this would be appreciated.

mx-moth commented on wagtailpolls/forms.py:
```diff
@@ -34,10 +38,10 @@ def recent_vote(self):
         if recent_vote is not None:
             if timezone.now() - recent_vote.time < datetime.timedelta(minutes=settings.VOTE_COOLDOWN):
-                self.add_error(None, 'Sorry, you cannot vote twice in %s minutes' % (settings.VOTE_COOLDOWN))
+                self.add_error(None, _('Sorry, you cannot vote twice in %s minutes') % (settings.VOTE_COOLDOWN))
```
You should use [`ungettext`](https://docs.djangoproject.com/en/1.10/topics/i18n/translation/#pluralization) for translations including plurals, although this is already broken in English when `VOTE_COOLDOWN = 1`

mx-moth commented on wagtailpolls/views/vote.py:
```diff
@@ -27,7 +28,7 @@ def vote(request, poll_pk):
 
     except ValueError:
-        return HttpResponse("<h1>Oops, there is no poll to vote on!</h1>", status=500)
+        return HttpResponse("<h1>{0}</h1>".format(_('Oops, there is no poll to vote on!')), status=500)
```
Error messages do not generally get translated. I would suggest this is better off as `raise Http404` as well, although that change is unrelated to translations.

mx-moth commented on wagtailpolls/templates/wagtailpolls/poll_list.html:
```diff
@@ -25,7 +25,7 @@
 						{% if poll.date_created %}
 						<div class="human-readable-date" title="{{ poll.date_created|date:"d M Y H:i" }}">
-						{{ poll.date_created|timesince }} ago</div>{% endif %}
+						{{ poll.date_created|timesince }} {% blocktrans %}ago{% endblocktrans %}</div>{% endif %}
```
For situations like this, where there is something like `<quantity> <modifier>`, it is best to translate the whole phrase, like:

``` html+django
{% blocktrans with when=poll.date_created|timesince %}{{ when }} ago{% endblocktrans %}
```

Not all languages follow the same structure as English in where they put modifiers for some types of measurements.

mx-moth commented on wagtailpolls/templates/wagtailpolls/poll_list.html:
```diff
@@ -25,7 +25,7 @@
 					<td class='created_on'>
 						{% if poll.date_created %}
 						<div class="human-readable-date" title="{{ poll.date_created|date:"d M Y H:i" }}">
```
There are a few formats that the `|date` filter understands that produce locale appropriate dates. Under the big table of format options, [the date docs](https://docs.djangoproject.com/en/1.10/ref/templates/builtins/#date) have more info. In this case, something like `{{ poll.date_created|date:"SHORT_DATETIME_FORMAT" }}` would be appropriate.

mx-moth commented on wagtailpolls/templates/wagtailpolls/poll_list.html:
```diff
@@ -25,7 +25,7 @@
+						{{ poll.date_created|timesince }} {% blocktrans %}ago{% endblocktrans %}</div>{% endif %}
 					</td>
 					<td class='children'>
 						<a href="{% url 'wagtailpolls_results' poll_pk=poll.pk %}" class="icon icon-arrow-right" title="Check out the poll results for '{{poll}}'"><b>Results</b></a>
```
You missed some strings here

mx-moth commented on wagtailpolls/templates/wagtailpolls/poll_list.html:
```diff
@@ -35,13 +35,13 @@
 </table>
 <div class="pagination">
-	<p>Page {{page.number}} of {{paginator.num_pages}}</p>
+	<p>{% blocktrans %}Page{% endblocktrans %} {{page.number}} {% blocktrans %}of{% endblocktrans %} {{paginator.num_pages}}</p>
```
Same comment as above regarding ordering of sentences like this. Something like:

``` html+django
<p>{% blocktrans with number=page.number total=paginator.num_pages %}Page {{ number }} of {{ total }}{% endblock %}</p>
```

mx-moth reviewed (changes requested):
You should set a translated `verbose_name` on all model fields that a user (admin or otherwise) might see:

``` python
class Poll(...):
    title = models.CharField(_('title'), max_length=128)
    # and so on ...
```

---

A thing some projects do, which you may want to use in the future: internal strings use 'single quotes', while user-facing strings use "double quotes". This makes looking for strings that need translation easier, as well as helping a developer understand the context of the string.

---

In the templates, `{% blocktrans %}` is used in a few places like `<title>` tags where `{% trans %}` would be better. It is just a bit less verbose, and more standard for short strings.

frague59 commented:
I'm working on in on my own fork...

LiamBrenner committed:
Implement feedback

Update use of trans tags

Update translation file

Fixed ungettext and 404 issues

SableWalnut commented:
@timheap Made the changes as mentioned in the review! :)

mx-moth commented on wagtailpolls/forms.py:
```diff
@@ -42,7 +43,7 @@ def clean(self):
         if recent_vote is not None:
             if timezone.now() - recent_vote.time < datetime.timedelta(minutes=cooldown):
-                self.add_error(None, _('Sorry, you cannot vote twice in %(cooldown)s minutes') % {'cooldown': cooldown})
+                self.add_error(None, __('Sorry, you cannot vote twice in %(cooldown)s minutes') % {'cooldown': cooldown})
```
This is incorrect. `ungettext` takes three arguments: the singular form, the plural form, and the count. Try:

``` python
__('Sorry, you can not vote twice in a minute',
   'Sorry, you can not vote twice in %(cooldown)s minutes',
   cooldown) % {'cooldown': cooldown})
```

mx-moth commented on wagtailpolls/views/vote.py:
```diff
@@ -33,7 +33,7 @@ def _vote(request, poll_pk):
         int(poll_pk)
     except ValueError:
-        return HttpResponse("<h1>{0}</h1>".format(_('Oops, there is no poll to vote on!')), status=500)
+        return Http404("<h1>{0}</h1>".format('Oops, there is no poll to vote on!'))
```
`raise Http404(...)`  or `return HttpResponseNotFound(...)`, but not `return Http404(...)`. [See the Django docs for more information](https://docs.djangoproject.com/en/1.10/topics/http/views/#the-http404-exception)

SableWalnut commented:
Fixed @timheap!

timheap commented:
Looks good to me now
